### PR TITLE
Support for Draft23 and remove of support for Draft 20

### PR DESCRIPTION
**Pull request by huitema**
Draft23 are just published. The changes are not big, as described in issues #647, #648, #649, #650, #652 and #653. All these issues are addressed in the PR. Issue #651 is not addressed, because it requires some clarification from the WG.

The PR also removes the hacks introduced to support draft-20 simultaneously with draft-22, and thus the support for draft 30.

### CODE REVIEW

File: picoquic/picoquic.vcxproj [hunk 1]
```diff
@@ -166,8 +166,6 @@
     <ClInclude Include="bytestream.h" />
     <ClInclude Include="democlient.h" />
     <ClInclude Include="demoserver.h" />
-    <ClInclude Include="fnv1a.h" />
```
Comment (steschu77) [inline picoquic/picoquic.vcxproj]: Was this intentional?

I like having the headers inside the solution as well to support "search in files" for these headers.

File: picoquic/logger.c [hunk 2]
```diff
@@ -784,13 +772,7 @@ size_t picoquic_log_generic_close_frame(picoquic_cnx_t * cnx, FILE* F, uint8_t*
     size_t l0 = 0;
 
     if (bytes_max >= 3) {
```
[the remark this answers is not in the document]
Comment (huitema) [reply picoquic/logger.c]: Should be 2 in fact...

File: picoquic/logger.c [hunk 3]
```diff
@@ -725,13 +719,7 @@ size_t picoquic_log_reset_stream_frame(picoquic_cnx_t * cnx, FILE* F, uint8_t* b
...
         byte_index += l1;
         if (l1 > 0 && bytes_max >= byte_index + 2) {
-            if (cnx != NULL && picoquic_supported_versions[cnx->version_index].version == PICOQUIC_TWELFTH_INTEROP_VERSION) {
-                error_code = PICOPARSE_16(bytes + byte_index);
-                l2 = 2;
-            }
-            else {
-                l2 = picoquic_varint_decode(bytes + byte_index, bytes_max - byte_index, &error_code);
-            }
+            l2 = picoquic_varint_decode(bytes + byte_index, bytes_max - byte_index, &error_code);
```
[the remark this answers is not in the document]
Comment (huitema) [reply picoquic/logger.c]: OK. But I actually added a better check on l1 and l2.

File: picoquic/picoquic.vcxproj (hunk 1 above)
Comment (huitema) [reply picoquic/picoquic.vcxproj]: fnv1a was removed from the solution. Democlient and Demoserver were moved to a different project. But removing bytestream is a mistake, thanks for noticing.

Comment (huitema) [reply picoquic/picoquic.vcxproj]: Actually, I should also remove democlient and demoserver, since they are not part of the main picoquic project.

Comment (huitema) [reply picoquic/picoquic.vcxproj]: Will leave fnv1a here for now, deal with removal later.

**Pull request merged**